### PR TITLE
*: Remove deprecated streaming

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/32679

Problem Summary:

Currently we have quite a few deprecated options with no plan to clean them up. This one is not known to confuse/affect users, but I thought I would take a look at the effort required to clean it up, since any simplification of variables helps usability.

It looks relatively straight forward, but we can reassess if feedback reports otherwise.

### Check List

Tests 

- [x] Unit test
Covered by existing tests, confirming removal is safe.
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility
Removes functionality from the server which was previously deprecated.

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
The deprecated system variable `tidb-enable-streaming` has been removed.
```

### CODE REVIEW

File: config/config_util.go [hunk 1]
```diff
@@ -53,7 +53,6 @@ var (
 		"Log.QueryLogMaxLen":              {},
 		"Log.ExpensiveThreshold":          {},
 		"CheckMb4ValueInUTF8":             {},
-		"EnableStreaming":                 {},
```
Comment (tiancaiamao) [inline config/config_util.go:56]: Shall we add it to here?
https://github.com/pingcap/tidb/blob/master/config/config.go#L781

Without this, if the old cluster use this configure, and upgrade a new version,
the tidb-server will not be able to run.

Comment (morgo) [reply config/config_util.go:56]: I don't believe the config parser is strict by default - but there is an optoin to do it: https://github.com/pingcap/tidb/pull/9855

But yes, good point. I will add it to the deprecated list.

Comment (morgo) [reply config/config_util.go:56]: I took a look. It's already there:

https://github.com/pingcap/tidb/blob/493eb45c75a8f9687d5bf9185e5e167aadc81012/config/config.go#L795

**Review by tiancaiamao — APPROVED** [review]
Please also check the document repo in case there is anything related to this feature.

File: distsql/distsql.go [hunk 2]
```diff
@@ -78,9 +78,7 @@ func Select(ctx context.Context, sctx sessionctx.Context, kvReq *kv.Request, fie
 		hook.(func(*kv.Request))(kvReq)
 	}
 
-	if !sctx.GetSessionVars().EnableStreaming {
-		kvReq.Streaming = false
-	}
+	kvReq.Streaming = false
```
Comment (bb7133) [inline distsql/distsql.go:81]: It looks like that `Streaming` field for `Request` can be removed as well? Anyway even if so, it doesn't have to be done in this PR.

**Review by bb7133 — APPROVED** [review]

**Pull request merged**